library/alloc/src/collections/btree/node.rs

```diff
@@ -365,11 +364,12 @@ impl<BorrowType, K, V, Type> NodeRef<BorrowType, K, V, Type> {
     ) -> Result<Handle<NodeRef<BorrowType, K, V, marker::Internal>, marker::Edge>, Self> {
         // We need to use raw pointers to nodes because, if BorrowType is marker::ValMut,
         // there might be outstanding mutable references to values that we must not invalidate.
-        let parent_as_leaf = unsafe { (*self.as_leaf_ptr()).parent as *const LeafNode<K, V> };
-        if let Some(non_zero) = NonNull::new(parent_as_leaf as *mut _) {
+        let leaf_ptr = self.as_leaf_ptr();
+        if let &Some(non_zero) = &unsafe { (*leaf_ptr).parent } {
```

Comment:
Hm, this looks a bit odd -- maybe we can drop the reference on both sides (in the pattern and before unsafe)?

Reply:
But then it's moved out of the option. Before this I had:
```if let Some(non_zero) = unsafe { (*leaf_ptr).parent.as_ref() } {```

Reply:
I think I prefer the latter form, it looks clearer to me.

Reply:
How about this fancy one?

Reply:
Also straigthened out the conversions to usize.